lib/payment.js

```diff
@@ -180,6 +182,7 @@ class Payment {
         // TODO: implement proper data validation before instantiation of new Payment
         const amount = MonetaryAmount.from(json);
         const p = new Payment(amount, json.sender.wallet, json.recipient.wallet, walletOrProvider);
+        _senderData.set(p, json.sender.data || '');
```

Comment:
This only enforces empty string when deserialized from JSON. If someone use the constructor it will still be undefined. See comment above.

Reply:
Good point 👍 
I made an update for both payment and receipt for this issue.